sql-plugin/src/main/scala/com/nvidia/spark/rapids/ColumnarOutputWriter.scala

```diff
@@ -67,14 +70,18 @@ abstract class ColumnarOutputWriter(context: TaskAttemptContext,
     rangeName: String,
     includeRetry: Boolean) extends HostBufferConsumer {
 
-  val tableWriter: TableWriter
-  val conf = context.getConfiguration
+  protected val tableWriter: TableWriter
 
-  private[this] val outputStream: FSDataOutputStream = {
+  protected val conf: Configuration = context.getConfiguration
+
+  protected def getOutputStream: FSDataOutputStream = {
     val hadoopPath = new Path(path)
     val fs = hadoopPath.getFileSystem(conf)
     fs.create(hadoopPath, false)
   }
+
+  protected val outputStream: FSDataOutputStream = getOutputStream
```

Comment:
Is there a reason to separate `outputStream` from `getOutputStream`?  One simply calls the other, and derived classes can override `outputStream` if necessary.  I don't actually see any overrides or accesses of this outside of this class, so wondering why this was made protected.
```suggestion
  protected val outputStream: FSDataOutputStream = {
    val hadoopPath = new Path(path)
    val fs = hadoopPath.getFileSystem(conf)
    fs.create(hadoopPath, false)
  }
```

Reply:
The reason for this was so that I could override the definition in the tests. I created a subclass of the `ColumnarOutputWriter` in the tests and mocked out `FSDataOutputStream`. I was not able to mock this with the val as it seemed that the act of extending `ColumnarOutputWriter` meant that the body of code (in your diff lines 78 to 80) needs to execute, even if the subclass is going to override `outputStream`. 

Overriding the def worked, and it called the subclass method instead.